internal/cmd/onepasswordtemplatefuncs.go

```diff
@@ -110,5 +121,79 @@ func getOnepasswordArgs(baseArgs, args []string) []string {
 	if len(args) > 2 {
 		baseArgs = append(baseArgs, "--account", args[2])
 	}
+
 	return baseArgs
 }
+
+// refreshSession will return the current session token if the token within the environment is still valid.
+// Otherwise it will ask the user to sign in and get the new token.
+func (c *Config) onepasswordGetOrRefreshSession(callerArgs []string) string {
+	var account string
+	if len(callerArgs) > 2 {
+		account = callerArgs[2]
+	}
+
+	// maybe there's already a valid token cached in this run for this account
+	token, ok := c.Onepassword.sessionTokens[account]
+	if ok {
+		return token
+	}
+
+	var args []string
+	if account == "" {
+		// If no account has been given then look for any session tokens you can find in the environment
+		token = onepasswordInferSessionToken()
```

Comment:
Instead of extracting a single session token, we iterate over all environment variables and populate `c.onepasswordConfig.sessionTokens` whenever we find an environment variable beginning with `OP_SESSION_`. This should simplify some of the overall logic.

Reply:
If you look closely, `c.onepasswordConfig.sessionTokens` only contains tokens that have been either verified to be valid, or freshly acquired, so we cannot pre-populate it without forcing refresh on every token in the environment, regardless of whether it used or not.

Reply:
Thanks for the explanation.